api/seedreconfig/seedreconfig.go

```diff
@@ -92,6 +92,41 @@ type SeedReconfiguration struct {
 
 	// PullSecret is the secret to use when pulling images. Equivalent to install-config.yaml's pullSecret.
 	PullSecret string `json:"pull_secret,omitempty"`
+
+	// Proxy is the proxy settings for the cluster. Equivalent to
+	// install-config.yaml's proxy. This will replace the proxy settings of the
+	// seed cluster. During IBI, the HTTP and HTTPS and NO proxy settings are
+	// given by the user, and IBIO will also calculate a final NO_PROXY
+	// configuration based on the user provided NO_PROXY and the cluster's
+	// configuration. During an IBU, this is simply taken from the cluster's
+	// Proxy CR .spec field.
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	// StatusProxy is the Proxy configuration from the Proxy CR's status field.
+	// The Cluster Network Operator calculates a proxy configuration based on
+	// the user provided proxy configuration (taken from the Proxy CR's spec)
+	// and puts that proxy configuration in the Proxy CR's status. The status
+	// proxy configuration is the one that's actually used by the operators in
+	// the cluster. During a seed reconfiguration, we need to know both the
+	// status proxy configuration and the user's spec proxy configuration, as
+	// both are required to perform a rollout free seed reconfiguration.
+	// Otherwise we would have to teach recert how to calculate the status
+	// proxy from the spec proxy which is non-trivial. During an IBU, we simply
+	// take this status from the cluster's Proxy CR .status field. During an
+	// IBI, IBIO should do its best to emulate the behavior of the CNO so that
+	// the status proxy configuration is identical to what CNO would have
+	// calculated, so that rollouts are avoided.
+	StatusProxy *Proxy `json:"status_proxy,omitempty"`
+
+	// The install-config.yaml used to generate the cluster. This is used to
+	// populate the cluster-config-v1 configmaps in the cluster, which usually
+	// hold a slightly modified version of the user's installation
+	// install-config.yaml. Changing this is important because the Cluster
+	// Network Operator uses the information in this configmap to set the Proxy
+	// CR's no_proxy status field. In IBU, LCA simply copies it from the
+	// upgraded cluster. In IBI, a fake install-config should be generated by
+	// the IBIO. This parameter is required when the proxy parameters are set.
+	InstallConfig string `json:"install_config,omitempty"`
```

Comment:
What recert will use from install-config? 
It seems strange for me to provide full install-config but on the other hand we can use it as api + new ibi specific fields. 
Currently it seems like we are going to provide duplicate values and it will be sort of confusing

Reply:
See comment

![image](https://github.com/openshift-kni/lifecycle-agent/assets/10882062/2022dbe3-1833-4b5e-a4e9-b147be319b78)

Reply:
>What recert will use from install-config?

Recert doesn't use it itself, it only updates the entirety of the install-config configmaps to match the install-config value you give it

Reply:
Which info it uses, which rollout we see?
In IBU it really doesn't matter but in IBIO this api is not something that can be usable in my opinion at least

Reply:
https://github.com/openshift/cluster-network-operator/blob/ce9414f20ccab2151786bcc0dab9fd0f213e3f75/pkg/util/proxyconfig/no_proxy.go#L63-L78

Reply:
>which rollout we see?

Reboot, because the no_proxy settings are taken from the install-config

Reply:
> https://github.com/openshift/cluster-network-operator/blob/ce9414f20ccab2151786bcc0dab9fd0f213e3f75/pkg/util/proxyconfig/no_proxy.go#L63-L78

In this case we can't actually take install config as is from SNO as it can differ from the seed, not sure we should add new seed requirements